canonical-web-and-design/maas-ui
# fix(ui): fix route rerenders

## Done

- Fix rerenders on all state changes due to using the [Route component prop](https://v5.reactrouter.com/web/api/Route/component). The relevant part of the docs being: `That means if you provide an inline function to the component prop, you would create a new component every render. This results in the existing component unmounting and the new component mounting instead of just updating the existing component.`

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the machine list.
- Tick some machines.
- Use the take action menu to open a form in the header. It should appear.

## Fixes

Fixes: #3533.
